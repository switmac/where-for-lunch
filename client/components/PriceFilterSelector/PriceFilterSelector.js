import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class PriceFilterSelector extends PureComponent {
  static propTypes = {
    action: PropTypes.func,
    filter: PropTypes.object,
  };

  handleOnClickAction = () => {
    const { filter } = this.props;
    this.props.action(filter);
  };

  render() {
    const { filter } = this.props;

    return (
      <div>
        <input type="checkbox" onClick={ this.handleOnClickAction } />
        { filter.label }
      </div>
    );
  }
}
