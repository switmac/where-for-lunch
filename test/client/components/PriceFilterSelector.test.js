
import React from 'react';
import { mount } from 'enzyme';
import PriceFilterSelector from 'client/components/PriceFilterSelector/PriceFilterSelector';
import renderer from 'react-test-renderer';


test('With Enzyme, PriceFilterSelector component without selecting a filter', () => {
  const filter = { id: 1, label: '$' };
  let selectedFilter;
  const testFxn = (value) => {
    selectedFilter = value;
  };
  testFxn.bind(this);
  mount(
    <PriceFilterSelector filter={ filter } action={ testFxn } />,
  );
  expect(selectedFilter).toEqual(undefined);
});

test('With Enzyme, simulate click to a PriceFilterSelector component', () => {
  const filter = { id: 1, label: '$' };
  let selectedFilter;
  const testFxn = (value) => {
    selectedFilter = value;
  };
  testFxn.bind(this);
  const component = mount(
    <PriceFilterSelector filter={ filter } action={ testFxn } />,
  );
  component.find('input').simulate('click');
  expect(selectedFilter).toEqual(filter);
});

test('With Jest, render PriceFilterSelector correctly', () => {
  const filter = { id: 1, label: '$' };
  const component = renderer
    .create(<PriceFilterSelector filter={ filter } />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
