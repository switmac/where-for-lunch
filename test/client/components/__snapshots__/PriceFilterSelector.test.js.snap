// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest, render PriceFilterSelector correctly 1`] = `
<div>
  <input
    onClick={[Function]}
    type="checkbox"
  />
  $
</div>
`;
